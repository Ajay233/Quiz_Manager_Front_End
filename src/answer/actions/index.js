import { getUsingParams, post, put, del } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { hideModal } from '../../modal/actions'
import { sessionExpired } from '../../utils/session'
import history from '../../history'

// create
export const addAnswer = (data, jwt) => {
  return (dispatch) => {
    return post("answer/create", [data], jwt).then((response) => {
      dispatch({
        type: "ADD_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer created", "success", true));
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        console.log(error.response);
        dispatch(setNotification("Error - unable to create answer", "error", true));
      }
    });
  }
}

// read
export const getAnswers = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("answer/findByQuestionId", param, jwt).then((response) => {
      dispatch({
        type: "SET_ANSWERS",
        payload: response.data
      });
    }).catch((error) => {
      dispatch(setNotification("Error retrieving answers", "error", true));
    });
  }
}

// update (there was a call to setCurrentAnswer but have removed this as it's not needed)
export const updateAnswer = (body, jwt) => {
  return (dispatch) => {
    return put("answer/update", [body], jwt).then((response) => {
      dispatch({
        type: "UPDATE_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer updated", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(setNotification("Error - unable to update answer", "error", true));
      }
    });
  }
}

// delete
export const deleteAnswer = (config, jwt) => {
  return (dispatch) => {
    return del("answer/delete", config, jwt).then((response) => {
      dispatch(hideModal())
      dispatch({
        type: "DELETE_ANSWER",
        payload: config.data[0]
      })
      dispatch(setNotification("Answer deleted", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(hideModal())
        dispatch(setNotification("Error - Unable to delete this answer", "error", true))
      }
    });
  }
}

export const setCurrentAnswer = (answer) => {
  return {
    type: "SET_CURRENT_ANSWER",
    payload: answer
  }
}
