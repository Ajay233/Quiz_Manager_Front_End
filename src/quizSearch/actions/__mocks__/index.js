export const setQuizes = jest.fn()

export const getAllQuizes = jest.fn()

export const getQuizSearchResults = jest.fn()

export const deleteQuiz = jest.fn()

export const addQuiz = jest.fn()

export const clearQuizes = jest.fn()
