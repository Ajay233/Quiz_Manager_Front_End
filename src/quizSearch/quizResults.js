import React from 'react'
import QuizResult from './quizResult'

const QuizResults = (props) => {

  const buildQuizes = () => {
    const {
      quizes,
      permission,
      jwt,
      setNotification,
      clearQuizes,
      setQuiz,
      showModal,
      showModal2,
      getQuestions,
      setQuizDownloadData
    } = props;
    const listOfQuizes = quizes.map(quiz => {
      return <QuizResult
                key={quizes.indexOf(quiz)}
                quiz={quiz}
                permission={permission}
                jwt={jwt}
                setNotification={setNotification}
                clearQuizes={clearQuizes}
                setQuiz={setQuiz}
                showModal={showModal}
                showModal2={showModal2}
                getQuestions={getQuestions}
                setQuizDownloadData={setQuizDownloadData}
              />
    })
    return listOfQuizes
  }



  const renderQuizes = () => {
    return props.quizes.length === 0 ? null : buildQuizes();
  }


  return(
    <div>
      {renderQuizes()}
    </div>
  );
}

export default QuizResults
