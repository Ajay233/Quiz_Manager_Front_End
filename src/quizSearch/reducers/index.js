export const setQuizSearchReducer = (state=[], action) => {
  switch (action.type) {
    // case "SET_QUIZES": return action.payload;
    case "GET_ALL_QUIZES": return action.payload;
    case "SET_QUIZ_SEARCH_RESULTS": return action.payload;
    case "DELETE_QUIZ": return state.filter(quiz => quiz !== action.payload);
    case "ADD_QUIZ": return state.length !== 0 && state[0].category === action.payload.category ? [...state, action.payload] : state;
    default: return state;
  }
}
