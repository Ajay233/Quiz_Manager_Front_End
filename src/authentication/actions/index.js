import { post } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import history from '../../history'

export const setUser = (endpoint, loginDetails) => {

  return (dispatch) => {

    post(endpoint, loginDetails).then((response) => {
      dispatch({
        type: "SET_USER_LOGGED_IN",
        payload: {
          id: response.data.user.id,
          forename: response.data.user.forename,
          surname: response.data.user.surname,
          email: response.data.user.email,
          permission: response.data.user.permission,
          verified: response.data.user.verified,
          jwt: response.data.jwt,
          loggedIn: true,
          response: response
        }
      })
      history.push("/")
      dispatch(setNotification(`Welcome back ${response.data.user.forename}`, "loginSuccess", true))
    }).catch((error) => {
      if(error.response.data === "NOT VERIFIED"){
        const verifyMsg = "Your email has not yet been veirified.  You will need to verify your email before you can log in"
        dispatch(setNotification(verifyMsg, "warning", true))
      } else {
        dispatch({
          type: "SET_USER_LOGGED_IN",
          payload: {
            error: error.response
          }
        });
        dispatch(setNotification("The username or password you entered was incorrect", "error", true));
      }
    })
  }
}

export const logOutUser = () => {
  return {
    type: "LOG_OUT_USER",
    payload: {
      forename: "",
      surname: "",
      email: "",
      permission: "",
      verified: "",
      jwt: "",
      loggedIn: false
    }
  }
}

export const setVerficationProcess = (endpoint, verificationDetails) => {

  return (dispatch) => {

    post(endpoint, verificationDetails).then((response) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "completed",
          token: "",
          error: {}
        }
      })
    }).catch((error) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "not verified",
          token: verificationDetails.token,
          error: error.response
        }
      })
    })
  }
}

export const setRedirect = (status) => {
  return {
    type: "SET_REDIRECT",
    payload: {
      status: status
    }
  }
}

export const logOut = () => {
  return {
    type: "RESET_APP"
  }
}

// export const getUsers = (endpoint, header) => {
//   return {
//     type: "GET_USER",
//     payload: {
//       userId
//   }
// }
