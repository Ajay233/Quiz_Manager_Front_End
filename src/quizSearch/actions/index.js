import { get, getUsingParams } from '../../axiosRequests/requests';
import { setNotification } from '../../notifications/actions'
import { sessionExpired } from '../../utils/session'

export const setQuizes = (quizes) => {
  return {
    type: "SET_QUIZES",
    payload: quizes
  }
}

export const getAllQuizes = (endpoint, token) => {
  return (dispatch) => {
    return get(endpoint, token).then((response) => {
      dispatch({
        type: "GET_ALL_QUIZES",
        payload: response.data
      })
    }).catch((error) => {
      console.log(error.response)
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(setNotification("Error - unable to get quiz list", "error", true));
      }

    });
  }
}

export const getQuizSearchResults = (endpoint, data, token, permission, errorMsg) => {
  return (dispatch) => {
    return getUsingParams(endpoint, data, token).then((response) => {
      let filteredQuizList = permission === "USER" ? response.data.filter(quiz => quiz.status === "READY") : response.data;
      dispatch({
        type: "SET_QUIZ_SEARCH_RESULTS",
        payload: filteredQuizList
      })
      if(filteredQuizList.length === 0){
         dispatch(setNotification(errorMsg, "error", true))
       }
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
      dispatch(setNotification(errorMsg, "error", true));
      }
    })
  }
}

export const deleteQuiz = (quiz) => {
  return {
    type: "DELETE_QUIZ",
    payload: quiz
  }
}

export const addQuiz = (quiz) => {
  return {
    type: "ADD_QUIZ",
    payload: quiz
  }
}

export const clearQuizes = () => {
  return {
    type: "CLEAR_QUIZZES_STORE",
    payload: []
  }
}

export const deleteQuizFromCategory = (quiz) => {
  return {
    type: "DELETE_QUIZ_FROM_CATEGORY",
    payload: quiz
  }
}
