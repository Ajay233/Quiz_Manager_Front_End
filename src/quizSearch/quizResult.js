import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import Modal from '../modal/modal'
import { setQuiz } from '../quiz/actions'
import { del } from '../axiosRequests/requests'
import { setNotification } from '../notifications/actions'
import { deleteQuiz } from './actions'
import { showModal } from '../modal/actions'

class QuizResult extends React.Component {

  handleEdit = () => {

  }

  renderModal = () => {

  }

  // handleDelete = () => {
  //   const { quiz, userData, deleteQuiz, setNotification } = this.props;
  //   const config = {
  //     data: quiz
  //   }
  //   del("quiz/delete", config, userData.jwt).then((response) => {
  //     deleteQuiz(quiz)
  //     setNotification("Quiz deleted", "success", true)
  //   }).catch((error) => {
  //     console.log(error.response)
  //     setNotification("Error - Unable to delete this quiz", "error", true)
  //   })
  // }

  handleDelete = () => {
    this.props.setQuiz(this.props.quiz);
    this.props.showModal();
  }

  handleView = () => {
    this.props.setQuiz(this.props.quiz);
  }

  handleStart = () => {
    console.log(this.props)
  }

  renderQuiz = () => {
    const { name, description} = this.props.quiz
    return(
      <div className="quiz">
        <div className="quizName">{name}</div>
        <div className="quizDescription">{description}</div>
        {this.renderOptions()}
      </div>
    );
  }

  // possible way to use different classes for option sheader if the header ends up being used
  optionsClass = (permission) => {
    return permission === "ADMIN" ? "" : "";
  }

  renderOptions = () => {
    const { permission } = this.props;
    return(
      <div className="options">
        { permission === "USER" ? <Link to="#" className="start" onClick={this.handleStart}><i className="far fa-play-circle blue"></i> Start</Link> : null }
        { permission === "READ-ONLY" ? <Link to="/viewQuiz" className="view" onClick={this.handleView}><i className="far fa-eye blue"></i> View</Link> : null }
        { permission === "ADMIN" ? <Link to="/editQuiz" className="edit" onClick={this.handleEdit}><i className="fas fa-edit blue"></i> Edit</Link> : null }
        { permission === "ADMIN" ? <Link to="#" className="deleteOption" onClick={this.handleDelete}><i className="fas fa-trash-alt red"></i> Delete</Link> : null }
      </div>
    );
  }

  render(){
    return(
      <div className="quizContainer">
        {this.renderQuiz()}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    userData: state.userData,
    modalState: state.showModal
  }
}

export default connect(mapStateToProps,{ setQuiz, setNotification, deleteQuiz, showModal })(QuizResult)
