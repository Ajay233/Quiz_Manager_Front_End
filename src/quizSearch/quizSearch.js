import React from 'react';
import { connect } from 'react-redux'
import QuizResults from './quizResults'
import QuizSearchByName from '../forms/findQuizByName'
import QuizSearchByCategory from '../forms/findQuizByCategory'
import Notification from '../notifications/notifications'
import Modal from '../modal/modal'
import { del } from '../axiosRequests/requests'
import { setNotification } from '../notifications/actions'
import { setQuizes, getQuizSearchResults, deleteQuiz, clearQuizes } from './actions'
import { getQuestions } from '../question/actions'
import { hideModal } from '../modal/actions'
import { sessionExpired } from '../utils/session'
import history from '../history'

import "../stylesheets/quizSearch.css"


export class QuizSearch extends React.Component {

  componentWillUnmount(){
    console.log("I will now clear the quizzes store :)")
    this.props.clearQuizes();
  }

  renderModal = () => {
    const { hideModal } = this.props
    const { showModal, showModal2 } = this.props.modalState
    if(showModal === true){
      return(
        <Modal
          type={"delete"}
          show={showModal}
          title={"Delete Quiz"}
          message={"You are about to delete a quiz which will also delete any questions and answers associated with is"}
          onDelete={this.handleDelete}
          onCancel={hideModal}
        />
      );
    } else if(showModal2 === true){
      return(
        <Modal
          type={"start"}
          show={showModal2}
          title={"Start Quiz"}
          message={"You are about to start a quiz, would you like to continue?"}
          onStart={this.handleStartQuiz}
          onCancel={hideModal}
        />
      );
    } else {
      return null;
    }
  }

  renderResultsTitle = (quizes) => {
    return quizes.length === 0 ? null : this.resultsTitle(quizes);
  }

  resultsTitle = (quizes) => {
    return  <div id="resultHeader">{`${quizes.length} ${quizes.length > 1 ? 'Results' : 'Result'}`}</div>
  }

  renderResultHeadings = (quizes) => {
    return quizes.length === 0 ? null : this.resultHeadings()
  }

// <div id="optionsHeader">Options</div>
  resultHeadings = () => {
    const { permission } = this.props.userData
    return(
      <div id="resultsHeadings">
        <div className={permission !== "USER" ? "nameHeader" : "nameHeaderExpanded"}>Quiz Name</div>
        <div className="descriptionHeader">Description</div>
        { permission !== "USER" ? <div className="statusHeader">Status</div> : null }
      </div>
    )
  }

  handleDelete = () => {
    const { quiz, userData, deleteQuiz, setNotification, hideModal } = this.props;
    const config = {
      data: quiz
    }
    del("quiz/delete", config, userData.jwt).then((response) => {
      hideModal()
      deleteQuiz(quiz)
      setNotification("Quiz deleted", "success", true)
    }).catch((error) => {
      console.log(error.response)
      if(error.response.status === 403){
        sessionExpired(this.props.dispatch);
      } else {
        hideModal()
        setNotification("Error - Unable to delete this quiz", "error", true)
      }
    })
  }

  handleStartQuiz = () => {
    const { userData, quiz, getQuestions, hideModal } = this.props
    const param = { quizId: quiz.id }
    getQuestions("question/findByQuizId", param, userData.jwt, true)
    hideModal()
    history.push("/startQuiz")
  }

  render(){
    const { quizes, userData, setNotification, getQuizSearchResults, lists } = this.props
    return(
      <div id="quizSearch">
        {this.renderModal()}
        <Notification />
        <div id="quizSearchTitle">
          <img id="searchTitleImg" src={require("../public/icons/search.png")} alt=""/>
          <div id="quizSearchTitleText">Quiz Search</div>
        </div>
        <div id="quizSearchForms">
          <div id="quizSearchByName">
            <QuizSearchByName
              jwt={userData.jwt}
              setNotification={setNotification}
              getQuizSearchResults={getQuizSearchResults}
              permission={userData.permission}
            />
          </div>
          <div id="divider-left"></div>
          <div id="divider-right"></div>
          <div id="quizSearchByCategory">
            <QuizSearchByCategory
              jwt={userData.jwt}
              setNotification={setNotification}
              getQuizSearchResults={getQuizSearchResults}
              categories={lists.categories}
              permission={userData.permission}
            />
          </div>
        </div>
        <div>
          {this.renderResultsTitle(quizes)}
          {this.renderResultHeadings(quizes)}
          <QuizResults quizes={quizes} permission={userData.permission} jwt={userData.jwt}/>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    userData: state.userData,
    notificationData: state.notificationData,
    quizes: state.quizes,
    quiz: state.quiz,
    lists: state.lists,
    modalState: state.modalState
  }
}

export default connect(mapStateToProps,
  {
    setQuizes,
    getQuizSearchResults,
    setNotification,
    hideModal,
    getQuestions,
    deleteQuiz,
    clearQuizes
  })(QuizSearch)
