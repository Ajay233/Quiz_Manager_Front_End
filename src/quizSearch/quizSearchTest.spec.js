import React from 'react'
import ReactDOM from 'react-dom'
import { QuizSearch, mapStateToProps } from './quizSearch'
import { Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render, fireEvent, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import history from '../history'

import { setNotification } from '../notifications/actions'
import { setQuizes, getQuizSearchResults, deleteQuiz, clearQuizes } from './actions'
import { getQuestions } from '../question/actions'
import { hideModal } from '../modal/actions'
import { sessionExpired } from '../utils/session'

jest.mock("../notifications/actions")
jest.mock("./actions")
jest.mock("../question/actions")
jest.mock("../modal/actions")
jest.mock("../utils/session")

const mockStore = configureStore({})

describe("mapStateToProps", () => {
  it("should map state to props", () => {
    const appState = {
      userData: { id: 1 },
      quiz: { id: 1, name: "test", description: "test" },
      modalState: { showModal: false, showModal2: false, showModal3: false },
      lists: ["Cat1", "Cat2", "Cat3"],
      notificationData: { message: "", type: "", show: false, timed: true },
      quizes: [
        { id: 1, name: 1, description: "test", category: "test", status: "READY" },
        { id: 2, name: 2, description: "test2", category: "test", status: "READY" }
      ]
    }

    const componentState = mapStateToProps(appState)
    expect(componentState).toEqual(appState)
  })
})

describe("QuizSearch", () => {

  let store;
  let modalState;
  let userData;
  let lists;
  let quizes;

  beforeEach(() => {
    store = mockStore({
      quiz: { id: 1, name: "test", description: "test" },
      lists: ["Cat1", "Cat2", "Cat3"],
      notificationData: {  },

    })

    modalState = { showModal: false, showModal2: false, showModal3: false }
    userData = { id: 1, jwt: "jwt", permission: "ADMIN", loggedIn: true }
    lists = { categories: ["test", "test2", "test3"] }
    quizes = [
      { id: 1, name: 1, description: "test", category: "test", status: "READY" },
      { id: 2, name: 2, description: "test2", category: "test", status: "READY" }
    ]

    ReactDOM.createPortal = jest.fn((element, node) => {
      return element
    })
  })

  it("should render quiz serchByName, searchByCategory forms and results", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <QuizSearch
            modalState={modalState}
            userData={userData}
            lists={lists}
            quizes={quizes}
            clearQuizes={clearQuizes}
            setNotification={setNotification}
          />
        </Router>
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })

  it("should render quiz serchByName, searchByCategory forms and no results", () => {
    const emptyQuizes = []
    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <QuizSearch
            modalState={modalState}
            userData={userData}
            lists={lists}
            quizes={emptyQuizes}
            clearQuizes={clearQuizes}
            setNotification={setNotification}
          />
        </Router>
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })

  describe("handleDelete", () => {

    let component;
    let quizToDelete;

    beforeEach(() => {
      const renderModal1 = { showModal: true, showModal2: false, showModal3: false }
      quizToDelete = { id: 1, name: "test", description: "test" }
      component = render(
        <Provider store={store}>
          <Router history={history}>
            <QuizSearch
              modalState={renderModal1}
              userData={userData}
              lists={lists}
              quizes={quizes}
              quiz={quizToDelete}
              clearQuizes={clearQuizes}
              hideModal={hideModal}
              deleteQuiz={deleteQuiz}
              setNotification={setNotification}
            />
          </Router>
        </Provider>
      )
    })

    it("should call hideModal, deleteQuiz and then setNotification", () => {
      fireEvent.click(component.getByTestId("modal-delete-button"))
      expect(deleteQuiz).toHaveBeenCalledTimes(1)
    })
  })

  describe("handleStartQuiz", () => {
    it("", () => {
      const renderModal2 = { showModal: false, showModal2: true, showModal3: false }
      const quiz = { id: 1, name: "test", description: "test" }
      const param = { quizId: 1 }
      const component = render(
        <Provider store={store}>
          <Router history={history}>
            <QuizSearch
              modalState={renderModal2}
              userData={userData}
              lists={lists}
              quizes={quizes}
              quiz={quiz}
              clearQuizes={clearQuizes}
              hideModal={hideModal}
              getQuestions={getQuestions}
              setNotification={setNotification}
            />
          </Router>
        </Provider>
      )

      history.push = jest.fn()

      fireEvent.click(component.getByTestId("modal-start-button"))
      expect(getQuestions).toHaveBeenCalledTimes(1)
      expect(getQuestions).toHaveBeenCalledWith("question/findByQuizId", param, "jwt", true)
      expect(hideModal).toHaveBeenCalledTimes(1)
      expect(history.push).toHaveBeenCalledTimes(1)
    })
  })

})
