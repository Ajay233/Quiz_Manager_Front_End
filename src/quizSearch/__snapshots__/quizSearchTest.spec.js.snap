// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizSearch should render quiz serchByName, searchByCategory forms and no results 1`] = `
<div
  id="quizSearch"
>
  <div
    id="quizSearchTitle"
  >
    <img
      alt=""
      id="searchTitleImg"
      src="search.png"
    />
    <div
      id="quizSearchTitleText"
    >
      Quiz Search
    </div>
  </div>
  <div
    id="quizSearchForms"
  >
    <div
      id="quizSearchByName"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div>
            <label>
              Quiz name:
            </label>
            <input
              className="inputBox"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              value=""
            />
          </div>
          <button
            className="submit"
          >
            <i
              className="fas fa-search"
            />
             Search
          </button>
        </form>
      </div>
    </div>
    <div
      id="divider-left"
    />
    <div
      id="divider-right"
    />
    <div
      id="quizSearchByCategory"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div>
            <label>
              Quiz category:
            </label>
            <select
              className="inputBox select"
              name="category"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              value=""
            >
              <option
                disabled={true}
                value=""
              >
                Select a category
              </option>
              <option
                value="test"
              >
                test
              </option>
              <option
                value="test2"
              >
                test2
              </option>
              <option
                value="test3"
              >
                test3
              </option>
            </select>
          </div>
          <button
            className="submit"
          >
            <i
              className="fas fa-search"
            />
             Search
          </button>
        </form>
      </div>
    </div>
  </div>
  <div>
    <div />
  </div>
</div>
`;

exports[`QuizSearch should render quiz serchByName, searchByCategory forms and results 1`] = `
<div
  id="quizSearch"
>
  <div
    id="quizSearchTitle"
  >
    <img
      alt=""
      id="searchTitleImg"
      src="search.png"
    />
    <div
      id="quizSearchTitleText"
    >
      Quiz Search
    </div>
  </div>
  <div
    id="quizSearchForms"
  >
    <div
      id="quizSearchByName"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div>
            <label>
              Quiz name:
            </label>
            <input
              className="inputBox"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              value=""
            />
          </div>
          <button
            className="submit"
          >
            <i
              className="fas fa-search"
            />
             Search
          </button>
        </form>
      </div>
    </div>
    <div
      id="divider-left"
    />
    <div
      id="divider-right"
    />
    <div
      id="quizSearchByCategory"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div>
            <label>
              Quiz category:
            </label>
            <select
              className="inputBox select"
              name="category"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              value=""
            >
              <option
                disabled={true}
                value=""
              >
                Select a category
              </option>
              <option
                value="test"
              >
                test
              </option>
              <option
                value="test2"
              >
                test2
              </option>
              <option
                value="test3"
              >
                test3
              </option>
            </select>
          </div>
          <button
            className="submit"
          >
            <i
              className="fas fa-search"
            />
             Search
          </button>
        </form>
      </div>
    </div>
  </div>
  <div>
    <div
      id="resultHeader"
    >
      2 Results
    </div>
    <div
      id="resultsHeadings"
    >
      <div
        className="nameHeader"
      >
        Quiz Name
      </div>
      <div
        className="descriptionHeader"
      >
        Description
      </div>
      <div
        className="statusHeader"
      >
        Status
      </div>
    </div>
    <div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            1
          </div>
          <div
            className="quizDescription"
          >
            test
          </div>
          <div
            className="quizStatus"
          >
            READY
          </div>
          <div
            className="options"
          >
            <a
              className="deleteOption"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fas fa-trash-alt red"
              />
               Delete
            </a>
            <a
              className="edit"
              href="/editQuiz"
              onClick={[Function]}
            >
              <i
                className="fas fa-edit blue"
              />
               Edit
            </a>
            <a
              className="start"
              href="/"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            2
          </div>
          <div
            className="quizDescription"
          >
            test2
          </div>
          <div
            className="quizStatus"
          >
            READY
          </div>
          <div
            className="options"
          >
            <a
              className="deleteOption"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fas fa-trash-alt red"
              />
               Delete
            </a>
            <a
              className="edit"
              href="/editQuiz"
              onClick={[Function]}
            >
              <i
                className="fas fa-edit blue"
              />
               Edit
            </a>
            <a
              className="start"
              href="/"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
